### PR TITLE
epee: fix some issues using connections after shutdown

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: contrib/epee/include/net/abstract_tcp_server2.inl [hunk 1]
```diff
@@ -431,19 +437,25 @@ PRAGMA_WARNING_DISABLE_VS(4355)
...
-    buffer_ssl_init_fill += bytes_transferred;
-    if (buffer_ssl_init_fill <= get_ssl_magic_size())
+    buffer_ssl_init_fill = bytes_transferred;
+    MTRACE("we now have " << buffer_ssl_init_fill << "/" << get_ssl_magic_size() << " bytes needed to detect SSL");
+    if (buffer_ssl_init_fill < get_ssl_magic_size())
     {
-      socket().async_receive(boost::asio::buffer(buffer_.data() + buffer_ssl_init_fill, buffer_.size() - buffer_ssl_init_fill),
+      // don't busy loop on this, ideally we'd want to queue a "async_receive_if_new_data" but there doesn't
+      // seem to be something like that in boost if we want to just peek at the data, so we'd need to copy and
+      // have a bit more code just for this. Though I'm just seeing async_read_until which might just work.
```
Comment (vtnerd) [inline contrib/epee/include/net/abstract_tcp_server2.inl:456]: `async_read_until` will not do what we need. Its going to read from the socket.

File: contrib/epee/include/net/abstract_tcp_server2.inl [hunk 2]
```diff
@@ -431,19 +437,25 @@ PRAGMA_WARNING_DISABLE_VS(4355)
...
       // offload the error case
       handle_read(e, bytes_transferred);
       return;
     }
 
-    reset_timer(get_timeout_from_bytes_read(bytes_transferred), false);
-
-    buffer_ssl_init_fill += bytes_transferred;
-    if (buffer_ssl_init_fill <= get_ssl_magic_size())
+    buffer_ssl_init_fill = bytes_transferred;
```
[the remark this answers is not in the document]
Comment (moneromooo-monero) [reply contrib/epee/include/net/abstract_tcp_server2.inl:450]: No, the data we read will come back on the next read, it's not removed.

[the remark this answers is not in the document]
Comment (vtnerd) [reply contrib/epee/include/net/abstract_tcp_server2.inl:450]: Then this member variable is unnecessary?

[the remark this answers is not in the document]
Comment (moneromooo-monero) [reply contrib/epee/include/net/abstract_tcp_server2.inl:450]: True, I'll remove it.

**Comment by Snipa22** [comment]
Merging as #7192 is merged

**Pull request merged**